// browserify-brunch/lib/browserify-instance.js
// Generated by CoffeeScript 1.10.0
var BrowserifyInstance, browserify, closeSync, mkdirp, openSync, path, ref, uglify, watchify, writeSync,
  bind = function(fn, me){ return function(){ return fn.apply(me, arguments); }; },
  hasProp = {}.hasOwnProperty;

browserify = require('browserify');

watchify = require('watchify');

mkdirp = require('mkdirp');

path = require('path');

ref = require('fs'), openSync = ref.openSync, writeSync = ref.writeSync, closeSync = ref.closeSync;

uglify = require('uglify-js');

BrowserifyInstance = (function() {
  function BrowserifyInstance(data) {
    var i, k, len, options, ref1, ref2, ref3, ref4, ref5, transform, v;
    this.data = data;
    this.handleUpdate = bind(this.handleUpdate, this);
    options = {
      debug: !this.data.main.production
    };
    ref1 = watchify.args;
    for (k in ref1) {
      if (!hasProp.call(ref1, k)) continue;
      v = ref1[k];
      options[k] = v;
    }
    ref2 = this.data.instanceOptions;
    for (k in ref2) {
      if (!hasProp.call(ref2, k)) continue;
      v = ref2[k];
      options[k] = v;
    }
    if (this.data.bundleOptions) {
      ref3 = this.data.bundleOptions;
      for (k in ref3) {
        if (!hasProp.call(ref3, k)) continue;
        v = ref3[k];
        options[k] = v;
      }
    }
    this.__w = browserify("./" + this.data.entry, options);
    if (this.data.main.watching) {
      this.__w = watchify(this.__w);
    }
    ref4 = this.data.transforms;
    for (i = 0, len = ref4.length; i < len; i++) {
      transform = ref4[i];
      this.__w.transform(transform);
    }
    if ((ref5 = this.data.onBrowserifyLoad) != null) {
      ref5.apply(this, [this.__w]);
    }
    if (this.data.main.watching) {
      this.__w.on('update', this.handleUpdate);
      this.handleUpdate();
    }
    null;
  }

  BrowserifyInstance.prototype.handleUpdate = function(fileContents, filePath, callback) {
    var ref1;
    this.running = true;
    if ((ref1 = this.data.onBeforeBundle) != null) {
      ref1.apply(this, [this.__w]);
    }
    return this.__w.bundle((function(_this) {
      return function(error, js) {
        var fd, minified, outputFile, ref2, ref3;
        if (error || (js == null)) {
          if (!_this.data.main.watching) {
            throw error;
          }
          console.error('Browserify Error', error);
          if (typeof callback === "function") {
            callback(error || true, fileContents, filePath);
          }
          return;
        }
        js = js.toString();
        if (_this.data.main.production) {
          minified = uglify.minify(js, {
            fromString: true
          });
          js = minified.code;
        }
        outputFile = _this.data.main.publicPath + "/" + _this.data.compiledPath;
        mkdirp.sync(path.dirname(outputFile));
        fd = openSync(outputFile, 'w+');
        writeSync(fd, js);
        closeSync(fd);
        _this.running = false;
        if ((ref2 = _this.data.onAfterBundle) != null) {
          ref2.apply(_this, arguments);
        }
        if ((ref3 = _this.data.main.__autoReloadServer) != null) {
          ref3.sendMessage('page');
        }
        return typeof callback === "function" ? callback(error, fileContents, filePath) : void 0;
      };
    })(this));
  };

  return BrowserifyInstance;

})();

module.exports = {
  BrowserifyInstance: BrowserifyInstance
};
