// cozydb/utils/errors.js
(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
// Generated by CoffeeScript 1.8.0
(function() {
  exports.NotFound = function(what) {
    var err, msg;
    msg = "" + what + " not found";
    err = new Error(msg);
    err.status = 404;
    return err;
  };

  exports.DSError = function(originalErr) {
    var err, msg;
    msg = "DSError : " + originalErr.message;
    err = new Error(msg);
    err.stack = originalErr.stack;
    err.status = 500;
    return err;
  };

  exports.WrongShemaError = function(msg) {
    var err;
    msg = "WrongShemaError: " + msg;
    err = new Error(msg);
    return err;
  };

  exports.NotOnNewModel = function() {
    return new Error("Wrong Usage : you attempted to call an instance method on a model\nwithout id.");
  };

}).call(this);

},{}]},{},[1]);
